### PR TITLE
add zabbix modules

### PR DESCRIPTION
**Pull request by cove**
add zabbix modules for groups, hosts screens and webchecks

### CODE REVIEW

File: monitoring/zabbix_group.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            module.exit_json(changed=False, result="No host group(s) to delete.")
+    else:
+        # create host groups
+        group_add_list = hostGroup.create_host_group(host_groups)
+        if len(group_add_list) > 0:
+            module.exit_json(changed=True, result="Successfully created host group(s): %s" % group_add_list)
+        else:
+            module.exit_json(changed=False)
+
+#<<INCLUDE_ANSIBLE_MODULE_COMMON>>
```
Comment (resmo) [inline monitoring/zabbix_group.py]: this boilerplate is deprecated (also see http://docs.ansible.com/developing_modules.html#common-module-boilerplate), you should use 

``` python
from ansible.module_utils.basic import *
main()
```

File: monitoring/zabbix_host.py [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+        if not interfaces or (interfaces and len(interfaces) == 0):
+            module.fail_json(msg="Specify at least one interface for creating host '%s'." % host_name)
+
+        # create host
+        host_id = host.add_host(host_name, group_ids, status, interfaces)
+        host.link_or_clear_template(host_id, template_ids)
+        module.exit_json(changed=True, result="Successfully added host %s (%s) and linked with template '%s'" % (
+            host_name, ip, link_templates))
+
+#<<INCLUDE_ANSIBLE_MODULE_COMMON>>
```
Comment (resmo) [inline monitoring/zabbix_host.py]: this boilerplate is deprecated (also see http://docs.ansible.com/developing_modules.html#common-module-boilerplate), you should use 

``` python
from ansible.module_utils.basic import *
main()
```

File: monitoring/zabbix_screen.py [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+    elif created_screens:
+        module.exit_json(changed=True, result="Successfully created screen(s): %s" % ",".join(created_screens))
+    elif changed_screens:
+        module.exit_json(changed=True, result="Successfully updated screen(s): %s" % ",".join(changed_screens))
+    elif deleted_screens:
+        module.exit_json(changed=True, result="Successfully deleted screen(s): %s" % ",".join(deleted_screens))
+    else:
+        module.exit_json(changed=False)
+
+# <<INCLUDE_ANSIBLE_MODULE_COMMON>>
```
Comment (resmo) [inline monitoring/zabbix_screen.py]: this boilerplate is deprecated (also see http://docs.ansible.com/developing_modules.html#common-module-boilerplate), you should use 

``` python
from ansible.module_utils.basic import *
main()
```

File: monitoring/zabbix_webcheck.py [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+        if not web_check_obj:
+            # create webcheck
+            web_check_class_obj.create_web_check(web_check_name, host_id, application_id, web_check_steps,
+                                                 scenario_optional_properties)
+        else:
+            # update webcheck
+            web_check_class_obj.update_web_check(web_check_obj, web_check_name, host_id, application_id,
+                                                 web_check_steps, scenario_optional_properties)
+
+# <<INCLUDE_ANSIBLE_MODULE_COMMON>>
```
Comment (resmo) [inline monitoring/zabbix_webcheck.py]: this boilerplate is deprecated (also see http://docs.ansible.com/developing_modules.html#common-module-boilerplate), you should use 

``` python
from ansible.module_utils.basic import *
main()
```

**Comment by cove** [comment]
thanks @resmo, updated.

**Comment by resmo** [comment]
@cove they merged my zabbix_group module. I am asking to revert it, so this can be pulled in.

File: monitoring/zabbix_webcheck.py [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+      web_check_name: "example Health Check"
+'''
+
+import json
+import base64
+import urllib2
+import random
+import time
+
+from ansible.module_utils.basic import *
```
Comment (resmo) [inline monitoring/zabbix_webcheck.py]: remove this line, duplicate of https://github.com/ansible/ansible-modules-extras/pull/44/files#diff-52e41b51661c5ff9d2854b88d297d200R391

File: monitoring/zabbix_screen.py [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+        state: present
+        graph_names:
+          - Example graph1
+          - Example graph2
+        graph_width: 200
+        graph_height: 100
+  when: inventory_hostname==groups['group_name'][0]
+'''
+
+from ansible.module_utils.basic import *
```
Comment (resmo) [inline monitoring/zabbix_screen.py]: remove this one

File: monitoring/zabbix_hostmacro.py [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+    login_password: password
+    host_name: ExampleHost
+    macro_name:Example macro
+    macro_value:Example value
+    state: present
+'''
+
+import logging
+import copy
+from ansible.module_utils.basic import *
```
Comment (resmo) [inline monitoring/zabbix_hostmacro.py]: remove this one

File: monitoring/zabbix_group.py [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+    login_user: username
+    login_password: password
+    state: present
+    host_groups:
+      - Example group1
+      - Example group2
+  when: inventory_hostname==groups['group_name'][0]
+'''
+
+from ansible.module_utils.basic import *
```
Comment (resmo) [inline monitoring/zabbix_group.py]: remove this one

**Comment by cove** [comment]
@resmo sounds good, thanks for helping review.

**Comment by bcoca** [comment]
Can you make this a separate request per module, its easier to review and add individually.

i'm guessing the rebase is needed because of the zabbix_group module we already merged 
@resmo I did not see your request to defer to these modules.

**Comment by cove** [comment]
re-submitted as #343, #344, #345 and #346 

NB: the zabbix_webcheck module is no longer needed in recent versions of zabbix, the zabbix_hostmacro module is used instead.

**Pull request closed without merging**